Source/Details/ASCollectionViewFlowLayoutCalculator.m

```diff
@@ -0,0 +1,35 @@
+//
+//  ASCollectionViewFlowLayoutCalculator.m
+//  AsyncDisplayKit
+//
+//  Created by Huy Nguyen on 28/2/17.
+//  Copyright © 2017 Facebook. All rights reserved.
+//
+
+#import <AsyncDisplayKit/ASCollectionViewFlowLayoutCalculator.h>
+#import <AsyncDisplayKit/ASCellNode.h>
+#import <AsyncDisplayKit/ASCollectionElement.h>
+#import <AsyncDisplayKit/ASElementMap.h>
+#import <AsyncDisplayKit/ASStackLayoutSpec.h>
+
+@implementation ASCollectionViewFlowLayoutCalculator
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)sizeRange forElementMap:(ASElementMap *)map
+{
+  NSMutableArray<ASCellNode *> *children = [NSMutableArray array];
```

Comment:
Can the map maintain this array internally?  These operations are pretty inefficient — especially the block-based iteration will create excessive retain / release operations, and also the array being created and destroyed for just one pass.  Imagine cases where there are hundreds of nodes for example, which is common.